### ISSUE TITLE
remote operators

### ISSUE THREAD

**Issue by axgkl** [jbaudanza/rxremote#2]
hey, cool project. question: ever thought about allowing to run operations like .filter on the server, declarable by the client ?

**Comment by jbaudanza** [comment]
Yes, I've given this some thought. I think there are two approaches:

1) Create some declarative query language, sort like MongoDb or firebase:

`observable.remoteFilter({price: {$lt: 20}})`

I built something this like for a related project, [rxeventstore](https://github.com/jbaudanza/jbaudanza/rxeventstore). See [filters.js](https://github.com/jbaudanza/rxeventstore/blob/master/src/database/filters.js)

2) Allow the client to execute javascript on the server.

`observable.remoteFilter('item => item.price < 10')`

Node comes with some pretty good [sandboxing functionality](https://nodejs.org/api/vm.html), but this approach would obviously come with a lot a security/DoS concerns.

I also think it would be interesting to consider how you might run operations like .scan and .reduce remotely on the server.

What do you think? Do you have a specific use case you're interesting in attacking?

**Comment by jbaudanza** [comment]
@AXGKl Is this still something you're interested in? If so, it would be good to discuss ideas about how it could work.
